pydy/viz/scene.py

```diff
@@ -154,6 +178,146 @@ def reference_frame(self, new_reference_frame):
         else:
             self._reference_frame = new_reference_frame
 
+    @property
+    def system(self):
+        return self._system
+
+    @system.setter
+    def system(self, new_system):
+
+        if new_system is not None and not isinstance(new_system, System):
```

Comment:
I had it that way at first and the tests errored. On construction of the instance system maybe be set to None, so this will error for no reason. Passing in None or a valid System are the two valid options.